src/libwrap/cdma_parse.c

```diff
@@ -0,0 +1,312 @@
+/*
+ * Copyright (c) 2012, 2013 Erik Faye-Lund
+ * Copyright (c) 2013 Avionic Design GmbH
+ * Copyright (c) 2013 Thierry Reding
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a
+ * copy of this software and associated documentation files (the "Software"),
+ * to deal in the Software without restriction, including without limitation
+ * the rights to use, copy, modify, merge, publish, distribute, sublicense,
+ * and/or sell copies of the Software, and to permit persons to whom the
+ * Software is furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL
+ * THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING
+ * FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER
+ * DEALINGS IN THE SOFTWARE.
+ */
+
+#include <stdio.h>
+
+#ifdef ENABLE_RNN
+#include <envytools/rnn.h>
+#include <envytools/rnndec.h>
+#endif
+
+#include "cdma_parse.h"
+
+enum nvhost_opcode {
+	NVHOST_OPCODE_SETCL,
+	NVHOST_OPCODE_INCR,
+	NVHOST_OPCODE_NONINCR,
+	NVHOST_OPCODE_MASK,
+	NVHOST_OPCODE_IMM,
+	NVHOST_OPCODE_RESTART,
+	NVHOST_OPCODE_GATHER,
+	NVHOST_OPCODE_EXTEND = 14,
+	NVHOST_OPCODE_CHDONE = 15,
+};
+
+#if 0
+static const char *nvhost_opcode_names[] = {
+	"NVHOST_OPCODE_SETCL",
+	"NVHOST_OPCODE_INCR",
+	"NVHOST_OPCODE_NONINCR",
+	"NVHOST_OPCODE_MASK",
+	"NVHOST_OPCODE_IMM",
+	"NVHOST_OPCODE_RESTART",
+	"NVHOST_OPCODE_GATHER",
+	"NVHOST_OPCODE_UNKNOWN_7",
+	"NVHOST_OPCODE_UNKNOWN_8",
+	"NVHOST_OPCODE_UNKNOWN_9",
+	"NVHOST_OPCODE_UNKNOWN_10",
+	"NVHOST_OPCODE_UNKNOWN_11",
+	"NVHOST_OPCODE_UNKNOWN_12",
+	"NVHOST_OPCODE_UNKNOWN_13",
+	"NVHOST_OPCODE_EXTEND",
+	"NVHOST_OPCODE_CHDONE",
+};
+#endif
+
+#define BIT(x) (1 << (x))
+
+struct nvhost_stream {
+	unsigned int num_words;
+	unsigned int position;
+	uint32_t *words;
+};
+
+static enum nvhost_opcode nvhost_stream_get_opcode(struct nvhost_stream *stream)
```

Comment:
This is not a big deal at all, but I think at this point it would make sense to rename these methods to cdma_foo rather than nvhost_foo, as they aren't really nvhost-specific anymore, no...

Reply:
ok